### PR TITLE
Add skip_train option for metrics

### PR DESCRIPTION
**Pull request by RakitinDen**
Added skip_train options for metrics in cross validation.

I hereby agree to the terms of the CLA available at https://yandex.ru/legal/cla/?lang=en.

### CODE REVIEW

File: catboost/libs/train_lib/cross_validation.cpp [hunk 1]
```diff
@@ -299,7 +303,6 @@ static void DisableMetricSkipTrain(NJson::TJsonValue* metric) {
     }
     hints["skip_train"] = "false";
     params["hints"] = MakeHintsDescription(hints);
```
Comment (annaveronika) [inline catboost/libs/train_lib/cross_validation.cpp]: DisableMetricSkipTrain is now not needed.

File: catboost/python-package/ut/medium/test.py [hunk 2]
```diff
@@ -1681,6 +1681,25 @@ def test_cv_custom_loss(task_type):
     assert "test-AUC-mean" in results
     return local_canonical_file(remove_time_from_json(JSON_LOG_PATH))
 
+def test_cv_skip_train(task_type):
```
Comment (annaveronika) [inline catboost/python-package/ut/medium/test.py]: Please also add test with AUC which has default skip_train=True and check that there is no train in results.

File: catboost/libs/train_lib/cross_validation.cpp [hunk 3]
```diff
@@ -423,11 +426,9 @@ void CrossValidate(
     // internal training output shouldn't interfere with main stdout
     updatedTrainOptionsJson["logging_level"] = "Silent";
 
-
-    // TODO(nikitxskv): Remove this hot-fix and make correct skip-metrics support in cv.
     DisableMetricsSkipTrain(&updatedTrainOptionsJson);
 
-
+    // TODO(nikitxskv): Remove this hot-fix and make correct skip-metrics support in cv.
```
Comment (annaveronika) [inline catboost/libs/train_lib/cross_validation.cpp]: This TODO should be removed

**Comment by annaveronika** [comment]
Thanks a lot for the changes! There were some updates in code, could you resolve the conflicts?

**Comment by andrey-khropov** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@andrey-khropov, internal review request created: 736861

**Pull request merged**